lib/cloudant.js

```diff
@@ -524,20 +531,40 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var resultArray = [];
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
-    if (docs.length === 0) return cb(null, {count: 0});
+    var conflicts = [];
+    if (docs.length === 0) {
+      err = new Error(g.f('No document found with filter: %j%s%s',
+        where, os.EOL, conflicts.join(os.EOL)));
+      err.statusCode = 404;
+      return cb(err);
+    }
+    _.each(docs, function(doc) {
+      if (data._rev) {
+        if (data._rev !== doc._rev) conflicts.push(doc._id);
```

Comment:
@jannyHou You are kind of correct. The condition I had there was wrong. Fixed it to account for both `id` and `_rev`. Tests should have been failing before. Should pass now (except the `updateAttributes` which still fails cause of juggler).

Doing bulk update directly was the way we previously did it. As discussed per the squad, we want to write the update functions so they work in this way:
- Use `updateAttributes` function to `update` a model instance.
- Use `update` function in `updateOrCreate` to update a model instance if exists.